Samples/UINamespace/uiNamespace.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+/**
+ * UINamespace Sample Extension
+ * 
+ * This sample extensions demonstrates how to use the UI namespace
+ * to create a popup dialog with additional UI that the user can interact with.
+ * The content in this dialog is actually an extension as well (see the 
+ * uiNamespaceDialog.js for details).  
+ * 
+ * This sample is an extension that auto refreshes datasources in the background of 
+ * a dashboard.  The extension has little need to take up any dashboard space, except
+ * when the user needs to adjust settings, so the UI namespace is used for that.
+ */
+
+// Wrap everything in an anonymous function to avoid polluting the global namespace
+(function () {
+  const defaultIntervalInMin = '5';
+  let refreshInterval;
+  let activeDatasourceIdList = [];
+
+  $(document).ready(function () {
+    tableau.extensions.initializeAsync().then(function() {
+      $('#play').click(configure);
+      $('#pause').click(pause);
+      
+      // This event allows for the parent extension and popup extension to keep their
+      // settings in sync.  This event will be triggered any time a setting is
+      // changed for this extension, in the parent or popup (i.e. when settings.saveAsync is called).
+      tableau.extensions.settings.addEventListener(tableau.TableauEventType.SettingsChanged, (settingsEvent) => {
+        updateExtensionBasedOnSettings(settingsEvent.newSettings)
+      });
+    });
+  });
+
+  function configure() {
+    const popupUrl = 'http://localhost:8765/Samples/UINamespace/uiNamespaceDialog.html';
```

Comment:
Should this use window.location to dynamically get this url?

Reply:
Good idea, will add.